manifests/enterprise/config.pp

```diff
@@ -0,0 +1,149 @@
+    file { '/etc/init.d/splunk':
+      ensure => 'absent',
+    }
+  }
+
+  if $facts['virtual'] == 'docker' {
+    ini_setting { 'OPTIMISTIC_ABOUT_FILE_LOCKING':
+      ensure  => present,
+      section => '',
+      setting => 'OPTIMISTIC_ABOUT_FILE_LOCKING',
+      value   => '1',
+      path    => "${splunk::enterprise::enterprise_homedir}/etc/splunk-launch.conf",
+    }
+  }
+
+  file { ["${splunk::enterprise::enterprise_homedir}/etc/system/local/alert_actions.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/authentication.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/authorize.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/deploymentclient.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/distsearch.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/indexes.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/inputs.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/limits.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/outputs.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/props.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/server.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/serverclass.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/transforms.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/ui-prefs.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/local/web.conf",
+          "${splunk::enterprise::enterprise_homedir}/etc/system/metadata/local.meta"]:
+    ensure => file,
+    tag    => 'splunk_enterprise',
+    owner  => $splunk::enterprise::splunk_user,
+    group  => $splunk::enterprise::splunk_user,
+    mode   => '0600',
+  }
+
+  if $splunk::enterprise::use_default_config {
+    splunk_input { 'default_host':
+      section => 'default',
+      setting => 'host',
+      value   => $splunk::enterprise::input_default_host,
+      tag     => 'splunk_server',
+    }
+    splunk_input { 'default_splunktcp':
+      section => "splunktcp://:${splunk::enterprise::logging_port}",
+      setting => 'connection_host',
+      value   => $splunk::enterprise::input_connection_host,
+      tag     => 'splunk_server',
+    }
+    splunk_web { 'splunk_server_splunkd_port':
+      section => 'settings',
+      setting => 'mgmtHostPort',
+      value   => "${splunk::enterprise::splunkd_listen}:${splunk::enterprise::splunkd_port}",
+      tag     => 'splunk_server',
+    }
+    splunk_web { 'splunk_server_web_port':
+      section => 'settings',
+      setting => 'httpport',
+      value   => $splunk::enterprise::web_httpport,
+      tag     => 'splunk_server',
+    }
+  }
+
+  File <| tag == 'splunk_enterprise' |>
```

Comment:
I'm wondering if we can simplify the whole block of class dependencies :thinking:

Reply:
I hope so, it was painful to write these out! I guess, technically, we could order the types to consolidate them into one block; order is arbitrary anyways:

File <| tag == 'splunk_enterprise' |>
-> Splunk_authentication   <| tag == 'splunk_enterprise' |>
-> Splunk_authorize        <| tag == 'splunk_enterprise' |>
-> Splunk_deploymentclient <| tag == 'splunk_enterprise' |>
... etc
~> Class['splunk::enterprise::service']

Reply:
@alexjfisher @binford2k can you have a look please?

Reply:
> Removed ability to include enterprise and forwarder on the same machine

Was this the only reason for tagging the resources?  Maybe much of this could be simplified??